modules/govuk_mysql/templates/usr/local/bin/xtrabackup_s3_restore.erb

```diff
@@ -5,42 +5,49 @@ exec 1> >(/usr/bin/logger -s -t $(basename $0)) 2>&1
 
 set -e
 
-DATE=$(date +%s)
-BASE_DIR=/tmp/mysql/basedir/$DATE/
-INCR_DIR=/tmp/mysql/incremental/$DATE/
+DATE=$(/bin/date +%F)
+CURRENT_RESTORE=/tmp/mysql/$DATE
+BASE_DIR=$CURRENT_RESTORE/basedir
+INCR_DIR=$CURRENT_RESTORE/incremental
 
 echo "Starting MySQL environment sync"
 
-# Check to see if Puppet is disabled, and skip the environment sync if it is
-# as we dumbly assume the server is under maintenance
-if test -f /var/lib/puppet/state/agent_disabled.lock; then
-  echo "Puppet is already disabled! Skipping environment sync and quitting"
-  exit 1
-else
-
-  govuk_puppet --disable "MySQL environment sync (by $0)"
+  /usr/local/bin/govuk_puppet -n mysql_restore -r "MySQL environment sync (by $0)" --disable
 
   # Create directories to restore into
-  for dir in "$BASE_DIR" "$INCR_DIR"; do
-    mkdir -p $dir
+  if [[ ! -d "$CURRENT_RESTORE" && ! -d "$BASE_DIR" && ! -d "$INCR_DIR" ]]; then
+    for dir in "$BASE_DIR" "$INCR_DIR"; do
+      /bin/mkdir $dir
+    done
+  else
```

Comment:
Reverted. Without the loop, the code looks less linear.  I added a comment for clarity.  Hope it reads well?

` /bin/rm -rf {$BASE_DIR/*,$INC_DIR/*}`